analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -183,7 +185,8 @@ private ProgramState ProcessBranchState(BasicBlock block, ControlFlowBranch bran
         if (branch.Source.BranchValue is { } branchValue && branch.Source.ConditionalSuccessor is not null) // This branching was conditional
         {
             state = SetBranchingConstraints(branch, state, branchValue);
-            state = checks.ConditionEvaluated(new(block, branchValue.ToSonar(), state, false, visitCount, lva.CapturedVariables));
+            SymbolicContext context = new(block, branchValue.ToSonar(), state, syntaxClassifier.IsInLoopCondition(branchValue.Syntax), loopDetector.IsInLoop(block), visitCount, lva.CapturedVariables);
```

Comment:
I changed it because it felt wrong to pass a default value for `inLoopCondition` but not for `isInLoop`. It doesn't really matter anyway because `inLoopCondition` gets removed in #8656.

This could lead to the question of why I did not want to pass a default value for `isInLoop`, so here are my thoughts:

Regarding UTs: No UT breaks because no implementation of `ConditionEvaluated` consumes `IsInLoop`.

Regarding performance: `syntaxClassifier.IsInLoopCondition(branchValue.Syntax)` has a potential performance impact. The loop detection, however, runs during the construction of `RoslynSymbolicExecution` and `loopDetector.IsInLoop(block)` is only a look-up without a significant performance impact.

Why did I not just pass `false` for `isInLoop`?
I find it plausible that in the future, an implementation of `ConditionEvaluated` will consume `isInLoop`. In that case it would be highly unexpected if it would just not work. Generally speaking, I don't think having a factually wrong state in the context is a good idea. This, combined with no performance impact, led me to the decision to pass the correct value here.

Reply:
The main problem I have with it is that if I delete it, nothing breaks. And having dead code is dangerous.

Reply:
I reverted `inLoopCondition` to `false` since it will be removed in the other PR regardless. I added a UT checking that `isInLoop` is passed correctly here.